### PR TITLE
(PUP-10672) add server_used report entry

### PR DESCRIPTION
**Pull request by ciprianbadescu**
it has the same value like `master_used`

### CODE REVIEW

File: lib/puppet/configurer.rb [hunk 1]
```diff
@@ -224,9 +224,10 @@ def run(options = {})
         if options[:catalog].nil? && do_failover
           server, port = find_functional_server
           if server.nil?
-            raise Puppet::Error, _("Could not select a functional puppet master from server_list: '%{server_list}'") % { server_list: Puppet.settings.value(:server_list, Puppet[:environment].to_sym, true) }
+            raise Puppet::Error, _("Could not select a functional puppet server from server_list: '%{server_list}'") % { server_list: Puppet.settings.value(:server_list, Puppet[:environment].to_sym, true) }
           else
             report.master_used = "#{server}:#{port}"
+            report.server_used = "#{server}:#{port}"
```
Comment (joshcooper) [inline lib/puppet/configurer.rb:230]: Should we remove the `Report#master_used` method and have `server_used` populate both entries in the report? Or did you want to save the removal of that method to Puppet 7?

Comment (ciprianbadescu) [reply lib/puppet/configurer.rb:230]: Yes ... I don't know why I have not done like this in the first time, sorry

File: lib/puppet/transaction/report.rb [hunk 2]
```diff
@@ -256,8 +256,8 @@ def initialize_from_hash(data)
     @time = data['time']
     @corrective_change = data['corrective_change']
 
-    if data['master_used']
-      @master_used = data['master_used']
+    if data['server_used']
+      @server_used = data['server_used']
     end
```
Comment (joshcooper) [inline lib/puppet/transaction/report.rb:261]: To preserve backwards compatibility when deserializing an older report, I think we want to check for `server_used`, but it if's not set, then fallback to `master_used`.

File: spec/integration/application/agent_spec.rb [hunk 3]
```diff
@@ -74,7 +75,8 @@
       end
 
       report = Puppet::Transaction::Report.convert_from(:yaml, File.read(Puppet[:lastrunreport]))
-      expect(report.master_used).to be_nil
+      expect(report.server_used).to be_nil
+      expect(report.server_used).to be_nil
```
Comment (GabrielNagy) [inline spec/integration/application/agent_spec.rb:79]: One of these should be deleted

File: spec/integration/application/agent_spec.rb [hunk 4]
```diff
@@ -91,7 +93,8 @@
          .and output(%r{Debug: Resolved service 'puppet' to https://127.0.0.1:#{port}/puppet/v3}).to_stdout
 
         report = Puppet::Transaction::Report.convert_from(:yaml, File.read(Puppet[:lastrunreport]))
-        expect(report.master_used).to eq("127.0.0.1:#{port}")
+        expect(report.server_used).to eq("127.0.0.1:#{port}")
```
Comment (GabrielNagy) [inline spec/integration/application/agent_spec.rb:96]: One of these should be deleted

File: lib/puppet/transaction/report.rb [hunk 5]
```diff
@@ -63,9 +63,9 @@ class Puppet::Transaction::Report
   # or 'on_failure'
   attr_accessor :cached_catalog_status
 
-  # Contains the name and port of the master that was successfully contacted
+  # Contains the name and port of the server that was successfully contacted
   # @return [String] a string of the format 'servername:port'
-  attr_accessor :master_used
+  attr_accessor :server_used
```
Comment (joshcooper) [inline lib/puppet/transaction/report.rb:68]: For backwards compatibility, maybe alias `master_used`?

```ruby
alias :master_used :server_used
alias :master_used= :server_used=
```

**Review by GabrielNagy — APPROVED** [review]

**Comment by joshcooper** [comment]
jenkins please test this with servertests

**Pull request merged**